api/envoy/extensions/clusters/dynamic_forward_proxy/v3/cluster.proto

```diff
@@ -32,4 +32,13 @@ message ClusterConfig {
   // in the :ref:`cluster's upstream_http_protocol_options
   // <envoy_v3_api_field_config.cluster.v3.Cluster.upstream_http_protocol_options>`
   bool allow_insecure_cluster_options = 2;
+
+  // If true allow HTTP/2 and HTTP/3 connections to be reused for requests to different
+  // origins than the connection was initially created for. This will only happen when the
+  // resolved address for the new connection matches the peer address of the connection and
+  // the TLS certificate is also valid for the new hostname. For example, if a connection
+  // has previously been established to foo.example.com at IP 1.2.3.4 with a certificate
+  // that is valid for *.example.com, then this connection could be used for requests to
+  // bar.example.com if that also resolved to 1.2.3.4.
```

Comment:
This is legit from RFC point of view but it's not how envoy likes.

The idea is `bar.example.com` is valid for "*.example.com" but it might not the upstream envoy wants.
Envoy may don't like "bar.example.com" even if the cert accept "*.example.com"

See https://github.com/envoyproxy/envoy/issues/6767#issuecomment-847534477

1. Can you add the the risk that when not to use `allow_coalesced_connections`? Otherwise, I would no brain turn on it when read your description.

2. It seems the implementation applies to forward proxy cluster but it's not clear in the doc here.

Reply:
1. Hm, this sounds like a good idea. I'm not quite sure how to phrase this though. I've taken a stab. Please take a look!
2. I'm not sure I understand. This is the doc for the forward cluster so I would have expected that it would be clear that the config described here is only for the cluster. Can you suggest some text to add to make this more clear?